core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/TransformCorrelatedGlobalAggregationWithProjection.java

```diff
@@ -175,16 +180,22 @@ public Result apply(CorrelatedJoinNode correlatedJoinNode, Captures captures, Co
         }
 
         source = decorrelatedSource.get().getNode();
+        Symbol nonNull = null;
 
-        // append non-null symbol on nested plan. It will be used to restore semantics of null-sensitive aggregations after LEFT join
-        Symbol nonNull = context.getSymbolAllocator().newSymbol("non_null", BOOLEAN);
-        source = new ProjectNode(
-                context.getIdAllocator().getNextId(),
-                source,
-                Assignments.builder()
-                        .putIdentities(source.getOutputSymbols())
-                        .put(nonNull, TRUE)
-                        .build());
+        AggregationNode globalAggregation = captures.get(AGGREGATION);
+        // bool_or aggregation is not null-sensitive so no need to add additional mask
+        boolean nonNullMaskNeeded = isNullMaskNeeded(globalAggregation);
```

Comment:
nit: inline `getBooleanAggregationSymbol(node).isEmpty()`. Then comment above is more direct